pageserver/src/tenant/timeline.rs

```diff
@@ -2946,7 +2946,7 @@ impl Timeline {
             let frozen_layer = Arc::clone(frozen_layer);
             move || {
                 // Write it out
-                let new_delta = frozen_layer.write_to_disk()?;
+                let new_delta = Handle::current().block_on(frozen_layer.write_to_disk())?;
```

Comment:
This block_on can be avoided easily: Move the line top the top of the function, before spawn_blocking().

That leaves just the par_fsync() stuff in the spawn_blocking block. And that can be changed into `par_fsync_async`, so that you don't need spawn_blocking() here at all.

Reply:
I think we should do this, but only shortly before we switch to async I/O, otherwise we'll be blocking executor threads. Commit 3693d1f431e89eec8231ab3ffa44021f1778d4cf has introduced the spawn_blocking here because the reads were blocking. They will still be even after this PR, even though they have `async` next to their name, as this PR is not switching the actual read calls sent to the OS to non-blocking ones.

Reply:
Ok. A comment would be nice, this looks pretty weird as it is.